packages/salesforce-adapter/src/metadata_deploy.ts

```diff
@@ -606,12 +609,22 @@ export const deployMetadata = async (
     deployedComponentsElemIdsByType,
     checkOnly ?? false,
   )
+
   const isSuccessfulChange = (change: Change<MetadataInstanceElement>): boolean => {
     const changeElem = getChangeData(change)
     const changeDeployedIds = changeToDeployedIds[changeElem.elemID.getFullName()]
     // TODO - this logic is not perfect, it might produce false positives when there are
     // child xml instances (because we pass in everything with a single change)
-    return successfulFullNames.some(successfulId => changeDeployedIds[successfulId.type]?.has(successfulId.fullName))
+    const metadataId = successfulFullNames.find(successfulId =>
+      changeDeployedIds[successfulId.type]?.has(successfulId.fullName),
+    )
+    if (metadataId) {
+      if (fetchProfile.isFeatureEnabled('shouldPopulateInternalIdAfterDeploy')) {
```

Comment:
```suggestion
    if (metadataId && fetchProfile.isFeatureEnabled('shouldPopulateInternalIdAfterDeploy')) {
```
Avoid nested ifs when possible